drivers/pancakeswap2.js

```diff
@@ -8,26 +8,127 @@ const { parseToFloat } = require('../lib/utils');
  * @augments Driver
  */
 class Pancakeswap2 extends Driver {
+  constructor() {
+    super({
+      supports: {
+        specificMarkets: true,
+      },
+    });
+  }
+
+  /**
+   * @param {Array} ids IDs of the pairs to retrieve
+   * @param {number|null} blockNumber Block number on which the pair data should be based
+   * @returns {Promise.Array} Array with the requested pairs
+   */
+  async getPairs(ids, blockNumber = null) {
+    // By default request the current top 300 markets with the highest volume (increasing the number
+    // beyond this causes timeouts when a block number is entered).
+    const selectQuery = ids ? `where: {id_in: ["${ids.join('", "')}"]}` : 'where: {volumeUSD_gt: 0}';
+    const blockQuery = blockNumber ? `block: {number: ${blockNumber}}` : '';
+
+    const { data: { pairs } } = await request({
+      method: 'POST',
+      url: 'https://bsc.streamingfast.io/subgraphs/name/pancakeswap/exchange-v2',
+      json: {
+        query: `
+          {
+            pairs(first: 300 ${selectQuery} ${blockQuery} orderBy: volumeUSD orderDirection: desc) {
```

Comment:
Previously the top 1000 were retrieved, should we increase the timeout of the request to make this possible with the new endpoint? Or should we use both the old and new endpoints, the old one for the top 1000 and the new one just for the specific markets?